components/tidb_query/src/lib.rs

```diff
@@ -18,6 +18,7 @@
 #![allow(redundant_semicolon)]
 // FIXME: ditto. probably a result of the above
 #![allow(clippy::no_effect)]
+#![allow(clippy::float_cmp)]
```

Comment:
Ah. I think allowing it for the whole tidb_query crate seems too rough. Can you make it effective only on that function?